src/gleam/int.gleam

```diff
@@ -515,17 +515,24 @@ fn do_undigits(
 }
 
 /// Generates a random int between the given minimum and maximum values.
-///
+/// if `inclusive < exclusive`, returns a value in `[inclusive, exclusive)`
+/// if `exclusive < inclusive`, returns a value in `(exclusive, inclusive]`
+/// if `inclusive == exclusive`, always returns `inclusive`
+/// May rarely return `exclusive` for the javascript target, due to floating point round errors
+/// 
 /// ## Examples
 ///
 /// ```gleam
 /// > random(1, 5)
 /// 2
 /// ```
-///
-pub fn random(min: Int, max: Int) -> Int {
-  float.random(to_float(min), to_float(max))
-  |> float.floor()
+@external(erlang, "gleam_stdlib", "random_int_range")
+pub fn random(inclusive: Int, exclusive: Int) -> Int {
+  let result = float.random(to_float(inclusive), to_float(exclusive))
+  case inclusive < exclusive {
+    True -> float.floor(result)
+    False -> float.ceiling(result)
+  }
```

Comment:
What does this change do? Seems like it is for something else.

Reply:
This is necessary for the case that exclusive < inclusive. For example if we do random(4, 0), we first get a float 4 >= x > 0. If we floor it, we will end up rounding many values to 0, which isn't allowed.

Reply:
"isn't allowed" is unclear phrasing. Without that check it will give the wrong result in the case of `exclusive< inclusive`.